### ISSUE TITLE
[OPS] Notify Moderators when a new file has been uploaded

### ISSUE THREAD

**Issue by alexxxmendonca** [pkp/pkp-lib#5975]
**Describe the problem you would like to solve**
Often times a manuscript needs adjustments on the PDF before making it public.
The authors can take up to several days before uploading a new file and the platform doesn't have an easy way of finding out if an author has transferred a new file to a submission.
Some authors use the Discussions to inform the upload of a new file, but not all authors do that.
As an Admin/Moderator, I have to access submissions one by one and check the activity log to check if an author uploaded a new file. It can be very difficult specially when dealing with a lot of submissions and it may lead to unwanted delays.

**Describe the solution you'd like**
Notify (via e-mail or via the notifications window, or both) when a new file was uploaded by an author.

**Who is asking for this feature?**
Admins and Moderators, for better control.

**Comment by ajnyga** [comment]
So basically if any file is uploaded after the actual submission process, there should be a notification? This sounds like something that all moderators do not want to happen. I mean in cases where not much moderation is happening and authors can just publish on their own. So it should be an optional feature.

@asmecher any preference on how to go forward with this: do I add a core feature that can be turned off if wanted?

I am not sure how easy it is to determine whether the file is uploaded during the submission process, but probably I can check the submission status to make sure. So technically not a difficult thing.

**Comment by alexxxmendonca** [comment]
From our experience's perspective, authors don't upload aditional files after submission.

Maybe the solution is not a notification for every single new file upload after submission, but currently there is no easy way of knowing when an author uploaded a revised version unless you access the submission. Sometimes you even have to check the history or download the file to check. This is very time consuming.

**Comment by alexxxmendonca** [comment]
_you request them to do changes to the galley and want to know when those changes are done_

That is mostly what happens.

Often times the authors need to fix the manuscript before we accept them. We requests these adjustments and we give them 2 working days to send a new version (within the same ID, not a new one). It would be good to be notified when they do it, it would speed up the process a lot.

**Comment by NateWr** [comment]
As @ajnyga has pointed out, we don't have a good high-level way to express an action like this. We only have the fine-grained activity log, and this is going to get very noisy if we start emailing moderators with every action that an author takes.

Instead, I'd suggest that we consider addressing this UX concern through the proposed activity feed (see https://github.com/pkp/pkp-lib/issues/5729). With the activity feed, a moderator would have a running record of activity across all of their assigned submissions, and could use this to keep on top of recent changes.

**Comment by alexxxmendonca** [comment]
> Instead, I'd suggest that we consider addressing this UX concern through the proposed activity feed (see #5729). 

That would still require accessing the submission, right?

What if submissions were flagged or highlighted in the queue whenever the author updated the metadata or uploaded a new file? We could then refer to the Activity feed to find out exactly what has been changed.

**Comment by NateWr** [comment]
> That would still require accessing the submission, right?

No. Take a look at the mockup in the linked issue. The idea is that users have access to an activity feed that includes log entries for all submissions that they are assigned to.

**Comment by alexxxmendonca** [comment]
Ah, ok. I did take a look at the mockup but I didn't pay attention to that.

Hopefully this will help solving this problem.

**Comment by asmecher** [comment]
(Here is a similar recent change to OJS 3.3.0: https://github.com/pkp/pkp-lib/issues/6643)

**Comment by asmecher** [comment]
Note that there's a similar requirement in the Open Research Europe project that we've met with OJS by implementing both upload capabilities (permitting an author to revise at any time) and a notification (to notify the editor when the author does so). It's currently coded as a plugin here: https://github.com/touhidurabir/oreWorkflow

**Comment by asmecher** [comment]
@YvesLepidus, if you're still interested in working on this one -- I've [separated it again](https://github.com/pkp/pkp-lib/issues/5729#issuecomment-4216125349) from #5729, which will get updated specs from Devika soon. It's more focused on log viewing, and you're more interested in notifications. If you're able, would you mind writing a few words about the approach you're interested in? It's been a long time since 2020 :)

**Comment by YvesLepidus** [comment]
Hi @asmecher 

> If you're able, would you mind writing a few words about the approach you're interested in? 

Our idea is to send a notification whenever an author makes any file revisions, which would include notifying when a file is edited or removed. There would also be an option to configure who receives these notifications/emails.

We are considering implementing this functionality as a plugin, as that may make more sense for this issue.
